qiskit_experiments/library/characterization/analysis/t1_analysis.py

```diff
@@ -67,3 +71,87 @@ def _evaluate_quality(self, fit_data: curve.CurveFitResult) -> Union[str, None]:
             return "good"
 
         return "bad"
+
+
+class T1KerneledAnalysis(curve.DecayAnalysis):
+    r"""A class to analyze T1 experiments with kerneled data.
+
+    # section: see_also
+        qiskit_experiments.curve_analysis.standard_analysis.decay.DecayAnalysis
+
+    """
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        """Default analysis options."""
+        options = super()._default_options()
+        options.curve_drawer.set_options(
+            xlabel="Delay",
+            ylabel="Normalized Projection on the Main Axis",
+            xval_unit="s",
+        )
+        options.result_parameters = [curve.ParameterRepr("tau", "T1", "s")]
+        options.normalization = True
+
+        return options
+
+    def _evaluate_quality(self, fit_data: curve.CurveFitResult) -> Union[str, None]:
+        """Algorithmic criteria for whether the fit is good or bad.
+
+        A good fit has:
+            - a reduced chi-squared lower than three
+            - absolute amp is within [0.9, 1.1]
+            - base is less than 0.1
+            - amp error is less than 0.1
+            - tau error is less than its value
+            - base error is less than 0.1
+        """
+        amp = fit_data.ufloat_params["amp"]
+        tau = fit_data.ufloat_params["tau"]
+        base = fit_data.ufloat_params["base"]
+
+        criteria = [
+            fit_data.reduced_chisq < 3,
+            abs(amp.nominal_value - 1.0) < 0.1,
+            abs(base.nominal_value) < 0.1,
+            curve.utils.is_error_not_significant(amp, absolute=0.1),
+            curve.utils.is_error_not_significant(tau),
+            curve.utils.is_error_not_significant(base, absolute=0.1),
+        ]
+
+        if all(criteria):
+            return "good"
+
+        return "bad"
+
+    def _format_data(
+        self,
+        curve_data: curve.CurveData,
+    ) -> curve.CurveData:
+        """Postprocessing for the processed dataset.
+
+        Args:
+            curve_data: Processed dataset created from experiment results.
+
+        Returns:
+            Formatted data.
+        """
+        # check if the SVD decomposition categorized 0 as 1 by calculating the average slope
+        diff_y = np.diff(unp.nominal_values(curve_data.y), axis=0)
+        avg_slope = sum(diff_y) / len(diff_y)
+        if avg_slope[0] > 0:
+            new_y_data = np.zeros(curve_data.y.shape)
+            for idx, y_data in enumerate(curve_data.y):
+                new_y_data[idx] = 1 - y_data
+
+            new_curve_data = CurveData(
+                x=curve_data.x,
+                y=new_y_data,
+                y_err=curve_data.y_err,
+                shots=curve_data.shots,
+                data_allocation=curve_data.data_allocation,
+                labels=curve_data.labels,
+            )
```

Comment:
In my opinion it's OK to change the `curve_data` that's in the input.
And then everything's that under the `if` can be probably modified to
```python
curve_data.y = 1 - curve_data.y
```

Reply:
It works! Changed as suggested.

Reply:
What do you mean that it's read-only? What happens when you try to overwrite it?

Reply:
I raise an error:
![image](https://user-images.githubusercontent.com/51112651/185890727-34c5a07f-58f2-47cd-a366-bc744feb9523.png)